packages/core/src/util/renderer.ts

```diff
@@ -454,10 +464,33 @@ export const mapStateToEnumControlProps = (
   ownProps: OwnPropsOfControl & OwnPropsOfEnum
 ): StatePropsOfControl & OwnPropsOfEnum => {
   const props: StatePropsOfControl = mapStateToControlProps(state, ownProps);
-  const options =
-    ownProps.options !== undefined
-      ? ownProps.options
-      : props.schema.enum || [props.schema.const];
+  const options: EnumOption[] =
+      ownProps.options ||
+      props.schema.enum?.map(enumToEnumOptionMapper) ||
+      props.schema.const && [enumToEnumOptionMapper(props.schema.const)];
+  return {
+    ...props,
+    options
+  };
+};
+
+/**
+ * Default mapStateToCellProps for enum control based on oneOf. Options is used for populating dropdown list
+ * @param state
+ * @param ownProps
+ * @returns {StatePropsOfControl & OwnPropsOfEnum}
+ */
+export const mapStateToOneOfEnumControlProps = (
+  state: JsonFormsState,
+  ownProps: OwnPropsOfControl & OwnPropsOfEnum
+): StatePropsOfControl & OwnPropsOfEnum => {
+  const props: StatePropsOfControl = mapStateToControlProps(state, ownProps);
+  const options: EnumOption[] =
+    ownProps.options ||
+    (props.schema.oneOf as JsonSchema[])?.map(e => ({
+      value: e.const,
+      label: e.title || typeof e.const === 'string' ? e.const : JSON.stringify(e.const)
```

Comment:
We need parenthesis here ;)
```suggestion
      label: e.title || (typeof e.const === 'string' ? e.const : JSON.stringify(e.const))
```